### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by maximst**
https://youtrack.raccoongang.com/issue/HBI-276

### CODE REVIEW

**Comment by raccoondeploy** [comment]
Build finished.

**Comment by raccoondeploy** [comment]
Refer to this link for build results (access rights to CI server needed): 

<a href='http://jenkins-ci.raccoongang.com/job/hawthorn-python-unittests-pr/24/'>http://jenkins-ci.raccoongang.com/job/hawthorn-python-unittests-pr/24/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/364/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/364/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/365/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/365/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/366/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/366/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/367/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/367/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/368/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/368/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/370/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/370/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/369/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/369/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/371/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/371/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/372/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/372/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/373/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/373/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/374/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/374/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/375/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/375/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/376/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/376/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/377/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/377/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/378/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/378/</a>
<a href='null'>null</a>

File: openedx/core/djangoapps/programs/tests/test_utils.py [hunk 1]
```diff
@@ -521,7 +521,7 @@ def test_simulate_progress(self, mock_completed_course_runs, mock_get_programs):
         program, program_uuid = data[0], data[0]['uuid']
         self._assert_progress(
             meter,
-            ProgressFactory(uuid=program_uuid, in_progress=1, not_started=1, grades={first_course_run_key: 0.0})
+            ProgressFactory(uuid=program_uuid, in_progress=1, not_started=0, grades={first_course_run_key: 0.0})
```
[the remark this answers is not in the document]
Comment (Marx86) [reply openedx/core/djangoapps/programs/tests/test_utils.py]: ይህ በ ‹PiProgressMeter ›ባደረግነው ለውጥ ምክንያት ነው ፡፡

File: pavelib/assets.py [hunk 2]
```diff
@@ -938,10 +938,14 @@ def update_assets(args):
...
     )
+    parser.add_argument(
+        '--disable-static-collector', action='store_true', dest='disable_static_collector', default=False,
+        help="Disable static collector"
+    )
     args = parser.parse_args(args)
 
     is_devstack = True
-    if args.settings in ('aws', 'static_collector'):
+    if not args.disable_static_collector and args.settings in ('aws', 'static_collector'):
```
Comment (idegtiarov) [inline pavelib/assets.py]: it looks that with aws and disable_static_collector args.settings we not only avoid adding  'static_collector' into settings but left is_devstack == True, it looks like a mistake or I've missed something?

This change affects the following changes.

File: common/djangoapps/entitlements/tests/test_tasks.py [hunk 3]
```diff
@@ -38,10 +38,11 @@ def test_checks_expiration(self, mock_datetime):
         """
         make_entitlement()
         make_entitlement()
+        init_call_count = mock_datetime.call_count
```
[the remark this answers is not in the document]
Comment (Marx86) [reply common/djangoapps/entitlements/tests/test_tasks.py]: В тесте проверяется сколько раз в таске expire_old_entitlements вызывается проперти expired_at_datetime. В данном тесте не важно сколько раз было вызвано это проперти при создании объекта при make_entitlement

File: lms/djangoapps/info_pages/models.py [hunk 4]
```diff
@@ -22,7 +22,7 @@ def get_pages():
 class InfoPage(TranslatableModel):
     PAGES = get_pages()
 
-    page = models.CharField(max_length=50, choices=PAGES, unique=True)
```
[the remark this answers is not in the document]
Comment (Marx86) [reply lms/djangoapps/info_pages/models.py]: Потому-что  список вариантов берется из сетингов, и может отличаться на разых инстансах. При этом создаются миграции из-за этого валится тест проверяющий миграции.

File: pavelib/assets.py (hunk 2 above)
Comment (Marx86) [reply pavelib/assets.py]: is_devstack как раз и выключает статик коллектор, она и была введена для этого. Это наши изменения.

File: lms/djangoapps/bulk_email/tests/test_email.py [hunk 5]
```diff
@@ -169,7 +169,14 @@ def mock_ugettext(text):
 
 
 @attr(shard=1)
-@patch.dict(settings.FEATURES, {'ENABLE_INSTRUCTOR_EMAIL': True, 'REQUIRE_COURSE_EMAIL_AUTH': False})
+@patch.dict(
+    settings.FEATURES,
+    {
+        'ENABLE_INSTRUCTOR_EMAIL': True,
+        'REQUIRE_COURSE_EMAIL_AUTH': False,
+        'BULK_EMAIL_FROM_DIFFERENT_ADDRESSES': True
```
[the remark this answers is not in the document]
Comment (Marx86) [reply lms/djangoapps/bulk_email/tests/test_email.py]: Ну это-же не кортеж

File: lms/djangoapps/bulk_email/tests/test_email.py [hunk 6]
```diff
@@ -195,7 +202,14 @@ def test_esperanto_platform(self):
...
```
[the remark this answers is not in the document]
Comment (Marx86) [reply lms/djangoapps/bulk_email/tests/test_email.py]: то-же

**Comment by raccoondeploy** [comment]
Build finished.

**Comment by raccoondeploy** [comment]
Refer to this link for build results (access rights to CI server needed): 

<a href='http://jenkins-ci.raccoongang.com/job/hawthorn-python-unittests-pr/25/'>http://jenkins-ci.raccoongang.com/job/hawthorn-python-unittests-pr/25/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/379/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/379/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/380/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/380/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/381/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/381/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/382/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/382/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/383/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/383/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/384/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/384/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/385/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/385/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/386/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/386/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/387/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/387/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/388/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/388/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/389/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/389/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/390/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/390/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/391/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/391/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/392/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/392/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/393/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/393/</a>

File: common/djangoapps/entitlements/tests/test_tasks.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (idegtiarov) [reply common/djangoapps/entitlements/tests/test_tasks.py]: got it, thanks for the explanation

File: lms/djangoapps/bulk_email/tests/test_email.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (idegtiarov) [reply lms/djangoapps/bulk_email/tests/test_email.py]: discussed locally

File: pavelib/assets.py (hunk 2 above)
Comment (idegtiarov) [reply pavelib/assets.py]: an additional comment would be very appreciated

**Comment by raccoondeploy** [comment]
Build finished.

**Comment by raccoondeploy** [comment]
Refer to this link for build results (access rights to CI server needed): 

<a href='http://jenkins-ci.raccoongang.com/job/hawthorn-python-unittests-pr/29/'>http://jenkins-ci.raccoongang.com/job/hawthorn-python-unittests-pr/29/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/451/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/451/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/452/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/452/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/453/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/453/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/454/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/454/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/455/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/455/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/456/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/456/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/457/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/457/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/458/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/458/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/459/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/459/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/460/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/460/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/461/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/461/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/462/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/462/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/463/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/463/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/464/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/464/</a>
<a href='http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/465/'>http://jenkins-ci.raccoongang.com/job/edx-platform-test-subset/465/</a>

**Pull request merged**